### ISSUE TITLE
Feature request: function_exists & class_exists

### ISSUE THREAD

**Issue by dmhendricks** [Kubitomakita/Requirements#7]
I know that there is the `custom_check` option, but it would be convenient to add checks for the existence (not just collisions) of classes or functions (particularly when they are expected to be loaded from dependencies and not plugins (else you'd just check for the plugin and/or version, or when a library isn't wrapped in a class (though rare)). Example:

```php
$requirements = new underDEV_Requirements( 'My Test Plugin', array(
   'php'        => '5.6',
   'function'   => array( 'is_gutenberg_page' ),  // Check if Gutenberg is loaded
   'class'      => array( '\\Aws\\Sdk' )          // Check if AWS SDK class exists
) );
```

Just an idea. I don't have time to implement at the moment, and it should probably reuse some code with the collision methods, but maybe eventually. Maybe take it one step further and make them associative arrays to make for a nicer error output:

```php
$requirements = new underDEV_Requirements( 'My Test Plugin', array(
   'php'        => '5.6',
   'function'   => array( 'Gutenberg' => 'is_gutenberg_page' ), // Check if Gutenberg is loaded
   'class'      => array( 'AWS SDK' => '\\Aws\\Sdk' )           // Check if AWS SDK class exists
) );
```

...

```php
foreach( $functions as $label => $func ) { 
   // ... if( !function_exists( $func ) {  
      _n( 'Missing dependency: %s', 'Missing dependencies: %s', $labels ),
   }
   // ...
}
```
etc, etc... Just spitballing...

**Comment by jakubmikita** [comment]
Very good idea Daniel!

I think we could take it another step further and besides checking for a function/class existence we could check for a function output, ie: `is_admin()`. It would disable the plugin silently on the frontend.

But... that will require the PHP 5.3 with closure support because making an eval is not a good idea.

Anyways, the idea is good and I'll definitely develop it sooner or later, but definitely wouldn't be able to do so this week.

**Comment by dmhendricks** [comment]
(When you get a chance, it would be great if you would create a new Release that includes those translation fixes that you merged to develop branch. I'm working on a new plugin, and for various reasons, I'd rather not set Composer's minimum-stability to dev. Thanks!)

**Comment by jakubmikita** [comment]
Hi Daniel, isn't it this one? https://github.com/Kubitomakita/Requirements/releases/tag/1.3.0

**Comment by dmhendricks** [comment]
I fixed the translation errors in PR#6, which was merged after the 1.3.0 release.

**Comment by jakubmikita** [comment]
And they are now included in 1.3.1 :)

**Comment by dmhendricks** [comment]
Thank you!
